packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -464,48 +456,14 @@ export function TransferPanelMain({
 
   useEffect(
     // Show on page load if SC wallet since destination address mandatory
-    () => setShowAdvancedSettings(isSmartContractWallet),
-    [isSmartContractWallet]
+    // or if destination address is provided
+    () =>
+      setShowAdvancedSettings(
+        isSmartContractWallet || !!destinationAddress || showAdvancedSettings
```

Comment:
Why do we have `showAdvancedSettings` as a dependency and condition for `setShowAdvancedSettings`?